### ISSUE TITLE
Too big to fork and start from scratch

### ISSUE THREAD

**Issue by ribtoks** [maxvoltar/photo-stream#47]
Hi

This is a really cool project, thanks for developing it!

There's a problem, however, when you try to set it up for yourself, the repo is too big. Probably it's because of the included photos (maybe in history too)? I expected a bunch of ruby and html/css files to be cloned in a fraction of a second, in reality I had to kill my `git clone` since it was taking too much time, downloading too much information to my hard drive.

<img width="492" alt="image" src="https://user-images.githubusercontent.com/505555/76486286-27359780-63dc-11ea-92ce-9ee1e69f8daa.png">

I propose to remove all heavy photos from history and not to include them in the repository. First action of any person who would set it up for themselves would be to remove those photos anyway.

**Comment by ribtoks** [comment]
Also - I halted repo as zip download from browser at 186MB - this is too much.

**Comment by chasebolt** [comment]
Best to migrate the images to git-lfs

**Comment by ribtoks** [comment]
@chasebolt if this repo is a building block to bootstrap own photo stream, best to not include any photos at all since you will have to remove them alltogether anyway to add your own photos.

**Comment by chasebolt** [comment]
I agree, but I assume he is using this repo for the demo site as well.

**Comment by maxvoltar** [comment]
1. Agree the repo size has gotten too big with all the old test photos in there. What's the easiest way to get rid of them?
2. Going forward, what would be the best approach to keep using this repo for the demo site (which is my own photo stream) without letting it get bloated. I'm prepared to run a certain command every couple of days 🙂

**Comment by msurguy** [comment]
@maxvoltar you could use Github actions to run commands automatically (and on a schedule). It is super easy! 
Adding certain folders to Gitignore might be best. You'd also probably need to rewrite history of the repo to strip out the photos from all commits

**Comment by maxvoltar** [comment]
Can't add the `/photos/original` folder to Gitignore since my photos need to make it to Netlify, which then resizes them. I _could_ change the way I roll out the website to build it on my machine first, then just push it to Netlify without build process, but that'd force me to always have the environment running (assuming I have my laptop with me). I like the idea of being able to add photos from anywhere while on the go.

**Comment by ribtoks** [comment]
@maxvoltar since this repo became popular, you may want to keep it for other people and forking purposes, just remove all images from history, `photos/originals` and all your personal settings (like domain) etc.

Meanwhile you can create another repo and host your own website with photos there just like you had till now. I think this will make more sense.

You can follow [this guide](https://help.github.com/en/github/authenticating-to-github/removing-sensitive-data-from-a-repository) in order to remove photos from history, particularly section that covers `git filter-branch` since this is a built-in you can use now.

**Comment by maxvoltar** [comment]
Issue: I can't fork my own repo's as far as I know. I'd like to keep improving the project, and pull those changes into the repo I use to serve my photo stream.

Also: How do I select all `.jpg`'s from the `/photos/originals`folder in order to delete them using that guide you linked to? I'm a `GitHub Desktop.app` user, and that's about the extend of what I know how to do 😄

**Comment by ribtoks** [comment]
@maxvoltar if I remember Jekyll/ruby correctly, you can have same repo, but git-ignored `config.yml` with your private settings (domain etc.). Also - git-ignore everything that has private settings (maybe leave an example file of that etc.)

Regarding `photos/originals`, try this:
- open terminal in the root of your repo
- Paste command `git filter-branch --index-filter "git rm -rf --cached --ignore-unmatch photos/originals" HEAD` and press Enter

After this you will need to do force push `git push -f origin master` since rewriting history is discouraged in general and git will not allow you to push new version to GitHub.

**Comment by maxvoltar** [comment]
Alright, older photos should now all be gone!

**Comment by ribtoks** [comment]
I still see few dozens of them in `photos/originals`.. Randomly opened photos are 4 and 3 MB. So its around 100MB of photos to pull

**Comment by petabyteboy** [comment]
Hi, for me the amount of download was about 300MB. I cloned the repo on a server, ran [bfg](https://rtyley.github.io/bfg-repo-cleaner/) and pushed it to [my fork](https://github.com/petabyteboy/photo-stream). How about using a seperate repository for the demo?

**Comment by maxvoltar** [comment]
@petabyteboy any clue how I could base my own website off of the demo repo? I'm new to OS 🙂

**Comment by maxvoltar** [comment]
@petabyteboy if you want I could also run bfg if you can help me with the correct commands.

**Comment by waschinski** [comment]
> How about using a seperate repository for the demo?

Another idea: Create a seperate branch `maxvoltar` and keep your photos there. It should be easy keeping that branch up-to-date from master and you would just have to git switch to that.

I would welcome the photos being removed from master as I actually have a PR prepared which would get rid of them anyway. 😄

**Comment by waschinski** [comment]
> Regarding `photos/originals`, try this:
> 
> * open terminal in the root of your repo
> * Paste command `git filter-branch --index-filter "git rm -rf --cached --ignore-unmatch photos/originals" HEAD` and press Enter
> 
> After this you will need to do force push `git push -f origin master` since rewriting history is discouraged in general and git will not allow you to push new version to GitHub.

You do not happen to know if that history is bloating docker images build from the repo too? For some reason those images are huge at about 1.8GB.

**Comment by waschinski** [comment]
Just wanted to drop that I managed to reduce the size of the docker image significantly.
